Models/TreeModel.cpp

```diff
@@ -308,6 +309,7 @@ buffers::TreeNode *TreeModel::duplicateNode(const buffers::TreeNode &node) {
 
 void TreeModel::removeNode(const QModelIndex &index) {
   if (!index.isValid()) return;
+  MainWindow::nodeResource.remove(index);
```

Comment:
I don't really like this coupling of the tree model to the main window. Please instead have the main window call this in a slot which it connects to the tree model removed signal. That way the tree model doesn't have to know anything about the main window and we can use it in other ways.

Reply:
Here's the signal: <https://doc.qt.io/qt-5/qabstractitemmodel.html#rowsRemoved>